starter-code/src/viking.js

```diff
@@ -1,11 +1,104 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+    this.health = health;
+    this.strength = strength;
+}
+
+Soldier.prototype.attack=function(){
+    return this.strength;
+}
+
+Soldier.prototype.receiveDamage=function(damage){
+    this.health -= damage;
+}
 
 // Viking
-function Viking() {}
+Viking.prototype=Object.create(Soldier.prototype);
+
+function Viking (name,health,strength){
+    this.name=name;
+    this.health=health;
+    this.strength=strength;
+}
+
+Viking.prototype.attack=function(){
+    return this.strength;
+}
+
+Viking.prototype.receiveDamage=function(damage){
+    this.health -= damage;
+    if (this.health > 0){
+        return this.name + " has received " + damage + " points of damage";
+    }
+    if (this.health <= 0){
+        return this.name + " has died in act of combat";
+    }
+}
+
+Viking.prototype.battleCry = function(){
+    return "Odin Owns You All!"
+}
 
 // Saxon
-function Saxon() {}
+Saxon.prototype=Object.create(Soldier.prototype);
+
+function Saxon (health,strength){
+    this.health=health;
+    this.strength=strength;
+}
+
+Saxon.prototype.attack=function(){
+    return this.strength;
+}
+
+Saxon.prototype.receiveDamage=function(damage){
+    this.health -= damage;
+    if (this.health> 0){
+        return "A Saxon has received " + damage + " points of damage";
+    }
+    if (this.health <= 0){
+        return "A Saxon has died in combat";
+    }
+}
+
 
 // War
-function War() {}
+function War() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+}
+
+War.prototype.addViking=function(Viking){
+    this.vikingArmy.push(Viking);
+}
+
+War.prototype.addSaxon=function(Saxon){
+    this.saxonArmy.push(Saxon);
+}
+
+War.prototype.vikingAttack = function(){
+    var attack = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)].receiveDamage(this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)].strength);
+    if (this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)].health <= 0){
+        this.saxonArmy.splice(0,1);
+    }
+    return attack;
+}
+
```

Comment:
Here, you vikingAttack and saxonAttack methods work, but there's two things:
1. you repeat the same long lines a couple times, which makes room for type errors
2. updating / modifying such code takes longer, because you need to makes the changes in several places instead of one
3. it's harder to read for someone that doesn't know your code, and making your functions easy to understand is a major key in teamworking with developers

A way to solve those problems, would be to capture all the elements you use inside properly-named variables so you know exactly what you're doing, and it's way clearer. Here's an example:
```javascript
War.prototype.vikingAttack = function() {
    
    var randomSaxNum = Math.floor( Math.random() * this.saxonArmy.length );
    var randomSaxon = this.saxonArmy[ randomSaxNum ];
    
    var randomVikNum = Math.floor( Math.random() * this.vikingArmy.length );
    var randomViking = this.vikingArmy[ randomVikNum ];

    var result = randomSaxon.receiveDamage( randomViking.attack() );

    if( randomSaxon.health <= 0 ) {
        this.saxonArmy.splice( randomSaxNum, 1 );
    }

    return result;
};
```

Last thing, in case the Viking / Saxon dies, you executed the splice on the index 0 every time. Which means if Viking / Saxon dies, you always splice the first in the array. The idea here is to splice the viking that just died, so you want to replace the 0 by the actual index in the armies array, as in the example I pasted above